scripts/globals/abyssea.lua

```diff
@@ -1,248 +1,307 @@
+            SOJOURN = SOJOURN + 1;
+        end
+        return SOJOURN;
+    elseif (Type == "FURTHERANCE") then
+        if (player:hasKeyItem(SCARLET_ABYSSITE_OF_FURTHERANCE)) then
+            FURTHERANCE = FURTHERANCE + 1;
+        end
+        if (player:hasKeyItem(SAPPHIRE_ABYSSITE_OF_FURTHERANCE)) then
+            FURTHERANCE = FURTHERANCE + 1;
+        end
+        if (player:hasKeyItem(IVORY_ABYSSITE_OF_FURTHERANCE)) then
+            FURTHERANCE = FURTHERANCE + 1;
+        end
+        return FURTHERANCE;
+    elseif (Type == "MERIT") then
+        if (player:hasKeyItem(AZURE_ABYSSITE_OF_MERIT)) then
+            MERIT = MERIT + 1;
+        end
+        if (player:hasKeyItem(VIRIDIAN_ABYSSITE_OF_MERIT)) then
+            MERIT = MERIT + 1;
+        end
+        if (player:hasKeyItem(JADE_ABYSSITE_OF_MERIT)) then
+            MERIT = MERIT + 1;
+        end
+        if (player:hasKeyItem(SAPPHIRE_ABYSSITE_OF_MERIT)) then
+            MERIT = MERIT + 1;
+        end
+        if (player:hasKeyItem(IVORY_ABYSSITE_OF_MERIT)) then
+            MERIT = MERIT + 1;
+        end
+        if (player:hasKeyItem(INDIGO_ABYSSITE_OF_MERIT)) then
+            MERIT = MERIT + 1;
+        end
+        return MERIT;
+    end
+end;
+
+-----------------------------------
+-- getDemiluneAbyssite
+-- returns total value of Demulune KeyItems
+-----------------------------------
+
+function getDemiluneAbyssite(player)
+    local Demilune = 0;
+    -- Todo: change this into proper bitmask
+    if (player:hasKeyItem(CLEAR_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 1;
+    end
+    if (player:hasKeyItem(COLORFUL_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 2;
+    end
+    if (player:hasKeyItem(SCARLET_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 4;
+    end
+    if (player:hasKeyItem(AZURE_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 8;
+    end
+    if (player:hasKeyItem(VIRIDIAN_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 16;
+    end
+    if (player:hasKeyItem(JADE_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 32;
+    end
+    if (player:hasKeyItem(SAPPHIRE_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 64;
+    end
+    if (player:hasKeyItem(CRIMSON_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 128;
+    end
+    if (player:hasKeyItem(EMERALD_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 256;
+    end
+    if (player:hasKeyItem(VERMILLION_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 512;
+    end
+    if (player:hasKeyItem(INDIGO_DEMILUNE_ABYSSITE)) then
+        Demilune = Demilune + 1024;
+    end
+    return Demilune;
+end;
+
+-----------------------------------
+-- prospectorEventFinish
+-- currently only handles items+tempItems
+-----------------------------------
+
+function prospectorEventFinish(player, option)
+    local itemMap =
+    {
+        [65537]  = {itemID = 11503, price = 4000, QTY = 1, TMP = false}, -- Perle Salade
```

Comment:
any time an event returns "strange" numbers, it's pretty much guaranteed to have some significance on the bit level